docs/examples/nmt/gnmt.md

```diff
@@ -0,0 +1,438 @@
+# Google NMT on IWSLT 2015 English-Vietnamese Translation
+
+In this notebook, we going to train Google NMT on IWSLT 2015 English-Vietnamese
+Dataset. The building prcoess includes: 1) load and process dataset and 2)
+create sampler and DataLoader and 3) build model and 4) write training epochs.
+
+## Load MXNET and Gluon
+
+```python
+import argparse
+import time
+import random
+import os
+import io
+import logging
+import numpy as np
+import mxnet as mx
+from mxnet import gluon
+from mxnet.gluon.data import ArrayDataset, SimpleDataset
+from mxnet.gluon.data import DataLoader
+import gluonnlp.data.batchify as btf
+from gluonnlp.data import ExpWidthBucket, FixedBucketSampler, IWSLT2015
+from gluonnlp.model import BeamSearchScorer
+from scripts.nmt.gnmt import get_gnmt_encoder_decoder
+from scripts.nmt.translation import NMTModel, BeamSearchTranslator
+from scripts.nmt.loss import SoftmaxCEMaskedLoss
+from scripts.nmt.utils import logging_config
+from scripts.nmt.bleu import compute_bleu
+import scripts.nmt._constants as _C
+```
+
+## Hyper-parameters
+
+```python
+np.random.seed(100)
+random.seed(100)
+mx.random.seed(10000)
+ctx = mx.gpu(0)
+
+# parameters for dataset
+dataset = 'IWSLT2015'
+src_lang = 'en'
+tgt_lang = 'vi'
+src_max_len = 50
+tgt_max_len = 50
+
+# parameters for model
+num_hidden = 512
+num_layers = 2
+num_bi_layers = 1
+dropout = 0.2
+
+# parameters for training
+batch_size = 128
+test_batch_size = 32
+num_buckets = 5
+epochs = 2
+clip = 5
+lr = 0.001
+lr_update_factor = 0.5
+log_interval = 10
+save_dir = 'gnmt_en_vi_u512'
+
+#parameters for testing
+beam_size = 10
+lp_alpha = 1.0
+lp_k = 5
+
+logging_config(save_dir)
+```
+
+## Load and Preprocess Dataset
+
+The following shows how to process the dataset and cache the processed dataset
+for the future use. The processing steps include: 1) clip the source and target
+sequences and 2) split the string input to a list of tokens and 3) map the
+string token into its index in the vocabulary and 4) append EOS token to source
```

Comment:
Since the notebook is for new comers, explaining abbr would help?
EOS token -> end-of-sentence (EOS) token.

Reply:
Sure